### PR TITLE
Portal map

### PR DESCRIPTION
**Pull request by MaXeraph**
- Portal Scene is created with ProBuilder
- Added ProBuilder package to project
- Portals will not work on Artillery projectiles

### CODE REVIEW

**Review by ikeia — APPROVED** [review]
I think it looks good, we should merge soon so I can use it in my level PR

File: Assets/Scripts/Portal.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	Vector3 distanceToExit;
+	void Start()
+	{
+		distanceToExit = portalOpposite.transform.position - transform.position;
+	}
+
+	void OnTriggerEnter(Collider c)
+	{
+		Debug.Log(c.gameObject.tag);
+		Debug.Log("distance: " + distanceToExit);
```
Comment (zhuandy42) [inline Assets/Scripts/Portal.cs:19]: we debugging

File: Assets/Scripts/Portal.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		if (c.gameObject.tag == "Player")
+		{
+			c.GetComponent<CharacterController>().enabled = false;
+			c.transform.position = c.transform.position + distanceToExit * 1.05f;
```
Comment (zhuandy42) [inline Assets/Scripts/Portal.cs:23]: is 1.05f so that we're slightly in front of the portal

File: Assets/Scripts/Portal.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+			c.GetComponent<CharacterController>().enabled = true;
+		}
+		else if (c.gameObject.tag == "Enemy")
+		{
+			c.GetComponent<EnemyBehaviour>().enabled = false;
+			c.GetComponent<NavMeshAgent>().SetDestination(portalOpposite.transform.position * 1.05f);
```
Comment (zhuandy42) [inline Assets/Scripts/Portal.cs:29]: is this setting the navmesh to move across the transparent bridge

**Review by zhuandy42 — APPROVED** [review]
lgtm, maybe comment out the debugs if they're useful to keep in for testing

File: Assets/Scripts/Portal.cs (hunk 2 above)
Comment (MaXeraph) [reply Assets/Scripts/Portal.cs:23]: yes

File: Assets/Scripts/Portal.cs (hunk 3 above)
Comment (MaXeraph) [reply Assets/Scripts/Portal.cs:29]: Not really, this was my attempt to make the enemy teleports, but with no success so far. Removed

**Review by ethanakester — APPROVED** [review]

**Pull request merged**